### ISSUE TITLE
user.dir is (unreachable)/ in root process

### ISSUE THREAD

**Issue by zhanghai** [topjohnwu/libsu#108]
Java NIO2 File API requires `System.getProperty("user.dir")` to return an absolute path (beginning with `/`) during its initialization, or it will throw an exception. However in the root process launched by this library, the value is `(unreachable)/` instead of `/`, and app code can't (easily) change it because JVM is making it read-only.

According to [documentation](https://developer.android.com/reference/java/lang/System#getProperties()), `user.dir` should always be `/` on Android.

I'm hitting this when trying to use Apache Commons Compress to read a ZIP file in the root process, and the library is trying to use the IO2 File API. This wasn't an issue for librootjava somehow.

**Comment by zhanghai** [comment]
The reason is that `getcwd()` in [` System_specialProperties`](https://cs.android.com/android/platform/superproject/+/master:libcore/ojluni/src/main/native/System.c;l=177;drc=c5f1494e3f0cdf6333bd0ba26bfac68da62e52d4) is returning `(unreachable)/`

From `man getcwd`:

```
       Following a change in Linux 2.6.36, the pathname returned by the
       getcwd() system call will be prefixed with the string
       "(unreachable)" if the current directory is not below the root
       directory of the current process (e.g., because the process set a
       new filesystem root using chroot(2) without changing its current
       directory into the new root).  Such behavior can also be caused
       by an unprivileged user by changing the current directory into
       another mount namespace.  When dealing with pathname from
       untrusted sources, callers of the functions described in this
       page should consider checking whether the returned pathname
       starts with '/' or '(' to avoid misinterpreting an unreachable
       path as a relative pathname.
```

So it seems to be related to mount namespace?

**Comment by zhanghai** [comment]
This is fixed in magisk 24.0 by https://github.com/topjohnwu/Magisk/commit/b2f152e6416042ae82bcaffb087081f3795ad4f6, but older magisk still has this issue. Maybe try to work around in libsu?

**Comment by zhanghai** [comment]
I worked around it in my own app, but may still be nice to have a workaround in libsu for this.
